src/lib/isFloat.js

```diff
@@ -5,7 +5,7 @@ export default function isFloat(str, options) {
   assertString(str);
   options = options || {};
   const float = new RegExp(`^(?:[-+])?(?:[0-9]+)?(?:\\${options.locale ? decimal[options.locale] : '.'}[0-9]*)?(?:[eE][\\+\\-]?(?:[0-9]+))?$`);
```

Comment:
```suggestion
  // largely derived from ECMAScript, with optional leading sign and allowing unnecessary leading zeroes
  // but not underscore separators
  // https://tc39.es/ecma262/multipage/ecmascript-language-lexical-grammar.html#prod-DecimalLiteral
  const float = new RegExp(`^(?:[-+])?(?:[0-9]+|(?=.[0-9]))(?:\\${options.locale ? decimal[options.locale] : '.'}[0-9]*)?(?:[eE][-+]?[0-9]+)?$`);
```
(and if you take this suggestion, then the following `if` is unnecessary)